odoo/addons/test_lint/tests/test_docstring.py

```diff
@@ -0,0 +1,144 @@
+import docutils
+import inspect
+
+from odoo.modules.registry import Registry
+from odoo.tests.common import get_db_name, tagged
+from odoo.addons.test_lint.tests.lint_case import LintCase
+
+RST_INFO_FIELDS_DOC = (
+    "https://www.sphinx-doc.org/en/master/usage/domains/python.html#info-field-lists")
+
+RST_INFO_FIELDS = {
+    'param': 'param',
+    'parameter': 'param',
+    'arg': 'param',
+    'argument': 'param',
+    'key': 'param',
+    'keyword': 'param',
+
+    'type': 'type',
+
+    'raises': 'raises',
+    'raise': 'raises',
+    'except': 'raises',
+    'exception': 'raises',
+
+    'var': 'var',
+    'ivar': 'var',
+    'cvar': 'var',
+
+    'vartype': 'vartype',
+
+    'returns': 'returns',
+    'return': 'returns',
+
+    'rtype': 'rtype',
+
+    'meta': 'meta',
+}
+
+
+def extract_docstring_types(method):
```

Comment:
Nitpick: move it to tools so that it can be used for rpc doc or leave a TODO for future.